src/main/java/kitchenpos/application/OrderService.java

```diff
@@ -1,109 +1,72 @@
-		}
-
-		final List<Long> menuIds = orderLineItems.stream()
-			.map(OrderLineItem::getMenuId)
-			.collect(Collectors.toList());
-
-		if (orderLineItems.size() != menuDao.countByIdIn(menuIds)) {
-			throw new IllegalArgumentException("주문 항목과 메뉴에서 찾은 주문항목의 수가 일치하지 않습니다.");
-		}
-
-		final OrderTable orderTable = orderTableDao.findById(order.getOrderTableId())
-			.orElseThrow(() -> new IllegalArgumentException("id에 해당하는 주문 테이블을 찾을 수 없습니다."));
-
-		if (orderTable.isEmpty()) {
-			throw new IllegalArgumentException("비어있는 테이블은 주문할 수 없습니다.");
-		}
-
-		order.setOrderTableId(orderTable.getId());
-		order.setOrderStatus(OrderStatus.COOKING.name());
-		order.setOrderedTime(LocalDateTime.now());
-
-		final Order savedOrder = orderDao.save(order);
-
-		final Long orderId = savedOrder.getId();
-		final List<OrderLineItem> savedOrderLineItems = new ArrayList<>();
-		for (final OrderLineItem orderLineItem : orderLineItems) {
-			orderLineItem.setOrderId(orderId);
-			savedOrderLineItems.add(orderLineItemDao.save(orderLineItem));
-		}
-		savedOrder.setOrderLineItems(savedOrderLineItems);
-
-		return savedOrder;
+	public OrderResponse create(final OrderRequest orderRequest) {
+		List<OrderLineItem> orderLineItems = createOrderLineItems(orderRequest);
+		final OrderTable orderTable = findOrderTable(orderRequest);
+		Order order = new Order(orderTable, OrderStatus.COOKING, LocalDateTime.now(), orderLineItems);
+		return OrderResponse.of(orderRepository.save(order));
 	}
 
-	public List<Order> list() {
-		final List<Order> orders = orderDao.findAll();
-
-		for (final Order order : orders) {
-			order.setOrderLineItems(orderLineItemDao.findAllByOrderId(order.getId()));
-		}
-
-		return orders;
+	public List<OrderResponse> list() {
+		return orderRepository.findAll().stream().map(OrderResponse::of).collect(Collectors.toList());
 	}
 
 	@Transactional
-	public Order changeOrderStatus(final Long orderId, final Order order) {
-		final Order savedOrder = orderDao.findById(orderId)
+	public OrderResponse changeOrderStatus(final Long orderId, final OrderRequest orderRequest) {
+		Order order = orderRepository.findById(orderId)
 			.orElseThrow(() -> new IllegalArgumentException("orderId에 해당하는 주문정보를 찾을 수 없습니다."));
+		order.changeState(orderRequest.getOrderStatus());
+		return OrderResponse.of(order);
+	}
 
-		if (Objects.equals(OrderStatus.COMPLETION.name(), savedOrder.getOrderStatus())) {
-			throw new IllegalArgumentException("이미 완료된 주문입니다.");
-		}
-
-		final OrderStatus orderStatus = OrderStatus.valueOf(order.getOrderStatus());
-		savedOrder.setOrderStatus(orderStatus.name());
-
-		orderDao.save(savedOrder);
+	private List<OrderLineItem> createOrderLineItems(OrderRequest orderRequest) {
+		return orderRequest.getOrderLineItems().stream()
+			.map(orderLineItemRequest -> {
+				Menu menu = findMenu(orderLineItemRequest.getMenuId());
+				return new OrderLineItem(null, menu, orderLineItemRequest.getQuantity());
```

Comment:
```suggestion
				return new OrderLineItem(menu, orderLineItemRequest.getQuantity());
```
생성자에서 2가지만 인자로 받아도 되지 않을까요

Reply:
다음 단계에서 반영하도록하겠습니다 😄